src/map/skill.c

```diff
@@ -1194,14 +1194,16 @@ static void skill_validate_autocast_data(struct map_session_data *sd, int skill_
 	// Determine if called by clif_parse_UseSkillMap().
 	bool use_skill_map = (skill_lv == 0 && (skill_id == AL_WARP || skill_id == AL_TELEPORT));
 
-	if (sd->auto_cast_current.type == AUTOCAST_NONE)
+	struct autocast_data auto_cast = sd->auto_cast_current;
```

Comment:
here will be copy of struct autocast_data from sd->auto_cast_current. This can be bit slow and allocate memory for copy. Is it really need here?

If dont need, you can get address of sd->auto_cast_current
``struct autocast_data *auto_cast = &sd->auto_cast_current;`` and use it like this ``auto_cast->type == ... ``

Reply:
Thanks for the hint. Fixed!